.github/workflows/check-pr.yaml

```
@@ -6,72 +6,43 @@ on:
     branches: [staging]
 
 jobs:
-  commits:
+  commitlint:
     runs-on: ubuntu-latest
     steps:
-      - name: format
-        if: always()
-        uses: taskmedia/action-conventional-commits@v1.1.23
+      - uses: actions/checkout@v4
         with:
-          types: "build|ci|docs|feat|fix|perf|refactor|style|test|revert|gomod|chore"
+          fetch-depth: 0
 
-      - name: length
-        if: ${{ github.actor != 'dependabot' && github.actor != 'dependabot[bot]' }}
-        uses: gsactions/commit-message-checker@v2
-        with:
-          pattern: '((^(?=(?:.|\n)*(?:^|\n)\[\d\]: .{69,}(?:$|\n)(?:.|\n)*)(?:.|\n)*$)|(^(?!(?:.|\n)*(?:^|\n).{74,}(?:$|\n)(?:.|\n)*)(?:.|\n)*$))'
-          flags: ''
-          error: 'The maximum line length of 74 characters is exceeded.'
-          excludeDescription: 'true'
-          excludeTitle: 'true'
-          checkAllCommitMessages: 'true'
-          accessToken: ${{ secrets.GITHUB_TOKEN }}
+      - name: Fetch commitlint config
+        run: |
+          wget "https://raw.githubusercontent.com/${GIT_REPO}/${GIT_REF}/commitlint.config.mjs"
+          if [ ! -f package.json ]; then
+            wget "https://raw.githubusercontent.com/${GIT_REPO}/${GIT_REF}/package.json"
+          fi
+          if [ ! -f package-lock.json ]; then
+            wget "https://raw.githubusercontent.com/${GIT_REPO}/${GIT_REF}/package-lock.json"
+          fi
+        env:
+          GIT_REPO: unikraft-cloud/x
+          GIT_REF: 146b50355461956ec4a9f1a54447fc29d8c633da
```

```suggestion
          GIT_REF: prod-staging
```

Should be updated when https://github.com/unikraft-cloud/x/pull/67 is merged.